src/main/java/org/kitteh/irc/client/library/feature/sts/STSPropertiesStorageManager.java

```diff
@@ -0,0 +1,206 @@
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.kitteh.irc.client.library.feature.sts;
+
+import org.kitteh.irc.client.library.exception.KittehSTSException;
+import org.kitteh.irc.client.library.util.StringUtil;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.Writer;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * Simple example implementation of an STSStorageManager.
+ */
+public class STSPropertiesStorageManager implements STSStorageManager {
+
+    public static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+    private final Properties properties = new Properties();
+    private final Writer writer;
+
+    /**
+     * Simple implementation of STSStorageManager which uses a properties file.
+     */
+    public STSPropertiesStorageManager(Writer writer, Reader reader) {
+        this.readData(reader);
+        this.writer = writer;
+    }
+
+    /**
+     * Reads in old data.
+     */
+    public void readData(Reader reader) {
+        try {
+            this.properties.load(reader);
+            reader.close();
+        } catch (IOException e) {
+            throw new KittehSTSException(e.getMessage());
+        }
+    }
+
+    /**
+     * Adds an STS policy to the store.
+     *
+     * @param hostname the hostname (as sent in the SNI by the client)
+     * @param duration the length (in seconds) until the expiry of this stored policy
+     * @param data all data sent by the server in the CAP LS "sts" value when connecting securely
+     */
+    @Override
+    public void addEntry(String hostname, long duration, Map<String, Optional<String>> data) {
+        this.properties.setProperty(hostname, getExpiryFromDuration(duration) + "; " + this.reserializeData(data));
+        this.saveData();
+    }
+
+    /**
+     * Saves data using the provided Writer.
+     */
+    private void saveData() {
+        try {
+            this.properties.store(this.writer, "This file contains all the gathered STS policies.");
+        } catch (IOException e) {
+            throw new KittehSTSException(e.getMessage());
+        }
+    }
+
+    /**
+     * Gets an STS policy from the store, looking it up via hostname.
+     *
+     * @param hostname the hostname (as sent in the SNI by the client)
+     * @return all data sent by the server in the CAP LS "sts" value when we connected securely
+     */
+    @Override
+    public Map<String, Optional<String>> getEntry(String hostname) {
+        this.pruneEntries();
+        if (!this.hasEntry(hostname)) {
+            return null;
```

Comment:
As discussed on IRC, the current use of `Optional` in arguments is less than ideal.